test/framework/ownerreference_helpers.go

```diff
@@ -0,0 +1,260 @@
+	},
+	machineHealthCheckKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK})
+	},
+}
+
+// Kind and GVK for types in the exp package.
+var (
+	clusterResourceSetKind        = "ClusterResourceSet"
+	clusterResourceSetBindingKind = "ClusterResourceSetBinding"
+
+	clusterResourceSetGVK = addonsv1.GroupVersion.WithKind(clusterResourceSetKind)
+)
+var ExpOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error{
+	clusterResourceSetKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{})
+	},
+	clusterResourceSetBindingKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK, clusterResourceSetGVK})
+	},
+}
+
+var (
+	configMapKind = "ConfigMap"
+	secretKind    = "Secret"
+)
+
+var SecretOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error{}
+var ConfigMapReferenceAssertions = map[string]func([]metav1.OwnerReference) error{
+	configMapKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterResourceSetGVK}, []schema.GroupVersionKind{})
+	},
+	secretKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{kubeadmControlPlaneGVK}, []schema.GroupVersionKind{kubeadmConfigGVK})
+	},
+}
+
+// Kind and GVK for types in the Kubeadm ControlPlane package.
+var (
+	kubeadmControlPlaneKind         = "KubeadmControlPlane"
+	kubeadmControlPlaneTemplateKind = "KubeadmControlPlaneTemplate"
+
+	kubeadmControlPlaneGVK = controlplanev1.GroupVersion.WithKind(kubeadmControlPlaneKind)
+)
+var KubeadmControlPlaneOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error{
+	kubeadmControlPlaneKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK})
+	},
+	kubeadmControlPlaneTemplateKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterClassGVK})
+	},
+}
+
+// Kind and GVK for types in the Kubeadm Bootstrap package.
+var (
+	kubeadmConfigKind         = "KubeadmConfig"
+	kubeadmConfigTemplateKind = "KubeadmConfigTemplate"
+
+	kubeadmConfigGVK = bootstrapv1.GroupVersion.WithKind(kubeadmConfigKind)
+)
+
+var KubeadmBootstrapOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error{
+	kubeadmConfigTemplateKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK}, []schema.GroupVersionKind{clusterClassGVK})
+	},
+	kubeadmConfigKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{machineGVK, machineSetGVK}, []schema.GroupVersionKind{machineGVK, kubeadmControlPlaneGVK})
```

Comment:
is the kubeadmConfigKind owned by a MachineSet? I have to double-check why this...
Same below for Docker machine owned by MachineSet/KCP